File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Script_outputTest.java
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

import org.junit.Test;

public class Script_outputTest {
  public static final String TYPE = "P2PKH";
  public static final String PUBKEYHASH = "2jmj7l5rSw0yVb-vlWAYkK-YBwk=";

  public static final Script_output SCRIPTTXOUT = new Script_output(TYPE, PUBKEYHASH);

  @Test
  public void testGetType() {
    assertEquals(TYPE, SCRIPTTXOUT.get_type());
  }

  @Test
  public void testGetPub_key_hash() {
    assertEquals(PUBKEYHASH, SCRIPTTXOUT.get_pubkey_hash());
  }

  @Test
  public void testEquals() {
    assertEquals(SCRIPTTXOUT, new Script_output(TYPE, PUBKEYHASH));
    String random = "random";
    assertNotEquals(SCRIPTTXOUT, new Script_output(random, PUBKEYHASH));
    assertNotEquals(SCRIPTTXOUT, new Script_output(TYPE, random));
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/ScriptOutputTest.kt
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash

import org.junit.Assert
import org.junit.Test

class ScriptOutputTest {
  @Test
  fun testGetType() {
    Assert.assertEquals(TYPE, SCRIPTTXOUT.type)
  }

  @Test
  fun testGetPub_key_hash() {
    Assert.assertEquals(PUBKEYHASH, SCRIPTTXOUT.pubKeyHash)
  }

  @Test
  fun testEquals() {
    Assert.assertEquals(SCRIPTTXOUT, ScriptOutput(TYPE, PUBKEYHASH))
    val random = "random"
    Assert.assertNotEquals(SCRIPTTXOUT, ScriptOutput(random, PUBKEYHASH))
    Assert.assertNotEquals(SCRIPTTXOUT, ScriptOutput(TYPE, random))
  }

  companion object {
    const val TYPE = "P2PKH"
    const val PUBKEYHASH = "2jmj7l5rSw0yVb-vlWAYkK-YBwk="
    val SCRIPTTXOUT = ScriptOutput(TYPE, PUBKEYHASH)
  }
}